constraint/pkg/client/client.go

```diff
@@ -363,18 +274,20 @@ func (c *Client) AddTemplate(templ *templates.ConstraintTemplate) (*types.Respon
 		return resp, nil
 	}
 
-	artifacts, err := c.createTemplateArtifacts(templ)
-	if err != nil {
-		return resp, err
-	}
-
 	c.mtx.Lock()
 	defer c.mtx.Unlock()
 
-	if err = c.backend.driver.PutModules(artifacts.namePrefix, artifacts.modules); err != nil {
-		return resp, fmt.Errorf("%w: %v", local.ErrCompile, err)
+	if d, ok := c.backend.driver.(interface{ AddExterns([]string) }); ok {
```

Comment:
When moving this to client.Init (see other review comments), instead cast to `local.Driver` and call this method. As-is we will not have a signal that this needs to be removed once AddExterns is removed from `local.Driver` (there won't be a compilation error to signal us that this is stale).

Reply:
We would have to export local.driver to allow the type cast, I can add error for the !ok  when casts the SetExterns

Reply:
I see two good ways to resolve this - feel free to use either. My preference is 1, but I don't have a strong opinion.

1. It should be fine to export `local.driver` (arguably, it should be exported). All of its members are private so there's no danger of callers messing with its fields.

2. Alternatively, define an interface in `local/local.go` which defines an interface with `AddExterns` as its only method. Add a line `var _ AddExternser = &driver{}` which will fail to compile when we remove the method from `local.go`. `client.go` should cast to this type to get the method.

Either of these solutions would work.